includes/html/pages/device/edit/snmp.inc.php

```diff
@@ -318,7 +318,7 @@
     <div class='form-group'>
     <label for='community' class='col-sm-2 control-label'>SNMP Community</label>
     <div class='col-sm-4'>
-    <input id='community' class='form-control' name='community' value='" . htmlspecialchars($device['community']) . "'/>
+    <input id='community' class='form-control' name='community' value='********' onfocus='this.value=(this.value==\"********\" ? decodeURIComponent(\"" . rawurlencode($device['community']) . "\") : this.value);'/>
```

Comment:
Change the type instead, using alpine it would be like
`<input type="password" value="hello" x-data="{ type: 'password' }" :type="type" @focus="type='text'">`

Reply:
Doesn't seem to work and still causes an XSS:

`<input type='password' value='" . $device['community'] . "' x-data='{ type: \'password\' }' :type='type' @focus='type=\'text\'' >`

Tried adding the onfocus back in but all that does is keep the output masked as `******`